auth/api/iam/api.go

```diff
@@ -398,24 +464,71 @@ func (r Wrapper) RequestAccessToken(ctx context.Context, request RequestAccessTo
 	if !isWallet {
 		return nil, core.InvalidInputError("did not owned by this node")
 	}
-	if request.Body.UserID != nil && len(*request.Body.UserID) > 0 {
-		// forward to user flow
-		return r.requestUserAccessToken(ctx, *requestHolder, request)
+	return requestHolder, nil
+}
+
+func (r Wrapper) RequestUserAccessToken(ctx context.Context, request RequestUserAccessTokenRequestObject) (RequestUserAccessTokenResponseObject, error) {
+	if request.Body == nil {
+		// why did oapi-codegen generate a pointer for the body??
+		return nil, core.InvalidInputError("missing request body")
+	}
+	requestHolder, err := r.extractWallet(ctx, request.Did)
+	if err != nil {
+		return nil, err
+	}
+
+	if request.Body.UserID == "" {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "missing userID")
+	}
+	// require RedirectURL
+	if request.Body.RedirectURL == "" {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "missing redirect url")
+	}
+
```

Comment:
missing check for `scope` and `redirectURL`?(but now we risk having the same checks again and again, everywhere)